packages/utils/keychain/src/index.ts

```diff
@@ -0,0 +1,73 @@
+import {
+	Ed25519Keypair,
+	X25519Keypair,
+	Keypair,
+	XSalsa20Poly1305,
+	Ed25519PublicKey,
+	X25519PublicKey,
+	ByteKey,
+	Secp256k1Keypair,
+	Secp256k1PublicKey,
+	PublicKeyEncryptionKey,
+	PublicSignKey
+} from "@peerbit/crypto";
+
+export type KeypairFromPublicKey<T> = T extends X25519PublicKey
+	? X25519Keypair
+	: T extends Ed25519PublicKey
+	? Ed25519Keypair
+	: T extends Secp256k1PublicKey
+	? Secp256k1Keypair
+	: T extends PublicSignKey | PublicKeyEncryptionKey
+	? Keypair
+	: never;
+
+export interface Keychain {
+	// Add a key to the keychain.
+	import(
+		parameters: (
```

Comment:
Maybe parameters should just be `Keypair` for asym keys and `Bytekey` for symmetric? If someone wants to pass in a specific key like x25519 they can still use the `X25519Keypair` and pass it in

Reply:
it might be possible to just have a "key" parameter and absorb everything with it. That will be easier to use from the user side. Just call

```js
await keychain.import(key, id?)  // optional id
```